aeron-client/src/main/java/io/aeron/protocol/ErrorFlyweight.java

```diff
@@ -0,0 +1,306 @@
+     *
+     * @return stream-id for the stream.
+     */
+    public int streamId()
+    {
+        return getInt(STREAM_ID_FIELD_OFFSET, LITTLE_ENDIAN);
+    }
+
+    /**
+     * Set stream-id for the stream.
+     *
+     * @param streamId stream-id for the stream.
+     * @return this for a fluent API.
+     */
+    public ErrorFlyweight streamId(final int streamId)
+    {
+        putInt(STREAM_ID_FIELD_OFFSET, streamId, LITTLE_ENDIAN);
+
+        return this;
+    }
+
+    /**
+     * The receiver-id for the stream.
+     *
+     * @return receiver-id for the stream.
+     */
+    public long receiverId()
+    {
+        return getLong(RECEIVER_ID_FIELD_OFFSET, LITTLE_ENDIAN);
+    }
+
+    /**
+     * Set receiver-id for the stream.
+     *
+     * @param receiverId receiver-id for the stream.
+     * @return this for a fluent API.
+     */
+    public ErrorFlyweight receiverId(final long receiverId)
+    {
+        putLong(RECEIVER_ID_FIELD_OFFSET, receiverId, LITTLE_ENDIAN);
+
+        return this;
+    }
+
+    /**
+     * Get the group tag for the message.
+     *
+     * @return group tag for the message.
+     */
+    public long groupTag()
+    {
+        return getLong(GROUP_TAG_FIELD_OFFSET, LITTLE_ENDIAN);
+    }
+
+    /**
+     * Determines if this message has the group tag flag set.
+     *
+     * @return <code>true</code> if the flag is set false otherwise.
+     */
+    public boolean hasGroupTag()
+    {
+        return HAS_GROUP_ID_FLAG == (HAS_GROUP_ID_FLAG & flags());
+    }
+
+    /**
+     * Set an optional group tag, null indicates the value should not be set. If non-null will set HAS_GROUP_TAG flag
+     * on the header. A null value will clear this flag and use a value of 0.
+     *
+     * @param groupTag optional group tag to be applied to this message.
+     * @return this for a fluent API.
+     */
+    public ErrorFlyweight groupTag(final Long groupTag)
+    {
+        if (null == groupTag)
+        {
+            flags((short)(~HAS_GROUP_ID_FLAG & flags()));
+        }
+        else
+        {
+            putLong(GROUP_TAG_FIELD_OFFSET, groupTag);
+            flags((short)(HAS_GROUP_ID_FLAG | flags()));
+        }
+
+        return this;
+    }
+
+    /**
+     * The error-code for the message.
+     *
+     * @return error-code for the message.
+     */
+    public int errorCode()
+    {
+        return getInt(ERROR_CODE_FIELD_OFFSET, LITTLE_ENDIAN);
+    }
+
+    /**
+     * Set error-code for the message.
+     *
+     * @param errorCode for the message.
+     * @return this for a fluent API.
+     */
+    public ErrorFlyweight errorCode(final int errorCode)
+    {
+        putInt(ERROR_CODE_FIELD_OFFSET, errorCode, LITTLE_ENDIAN);
+
+        return this;
+    }
+
+    /**
+     * Get the error string for the message.
+     *
+     * @return the error string for the message.
+     */
+    public String errorMessage()
+    {
+        return getStringUtf8(ERROR_STRING_FIELD_OFFSET);
```

Comment:
It would be better to treat encoded strings at US_ASCII as this is consistent elsewhere and ports to C.